node/src/reactor/participating.rs

```diff
@@ -737,6 +816,30 @@ impl reactor::Reactor for Reactor {
             chainspec.protocol_config.verifiable_chunked_hash_activation,
         )?;
 
+        let (chain_synchronizer, chain_synchronizer_effects) =
+            ChainSynchronizer::<ParticipatingEvent>::new(
+                chainspec.clone(),
+                config.node.clone(),
+                config.network.clone(),
+                next_upgrade_activation_point,
+                chain_sync_metrics,
+                effect_builder,
+            )?;
+        effects.extend(reactor::wrap_effects(
+            ParticipatingEvent::ChainSynchronizer,
+            chain_synchronizer_effects,
+        ));
```

Comment:
Can we skip all that if `config.node.sync_to_genesis == false`? I know that will require changes like making the `chain_synchronizer` in the `Reactor` an `Option` and probably some more, but maybe worth considering?

Reply:
I'd prefer to leave as is for now.  We've previously discussed having components able to be enabled or disabled dynamically, often when considering a single reactor model, and I'd like to leave such changes until then.